### PR TITLE
add Maven Wrapper support

### PR DESCRIPTION
**Pull request by vorburger**
see https://github.com/takari/maven-wrapper

### CODE REVIEW

**Comment by vorburger** [comment]
@rhuss & Co. how about this? 

I could use this to throw https://github.com/vorburger/opendaylight-bot on OpenShift...

PS: This will probably conflict with some of my other pending PRs; get those in first, then I'll rebase it.

**Comment by jamesnetherton** [comment]
I think we should only support this in the community images.

**Comment by vorburger** [comment]
@jamesnetherton so how exactly do we want to "condition" this? Similar to the Gradle "feature switch" in #121, so if I added a `withMavenWrapper` true/false to `images.yml` and `{{? fp.config.base.withMavenWrapper }}` in the template - would that work for you?

File: java/images/jboss/s2i/assemble [hunk 1]
```diff
@@ -137,14 +137,16 @@ function build_maven() {
 
   local mvn="mvn"
   if [ -f "mvnw" ]; then
+    echo "Using local mvnw"
+    ./mvnw --version
```
Comment (vorburger) [inline java/images/jboss/s2i/assemble]: this is kind of redundant with line 148 below, but never mind

**Comment by vorburger** [comment]
@jamesnetherton @rhuss done; `withMavenWrapper`, for now adds `./mvnw` only for community image.

**Pull request merged**